bin/oli/src/commands/cat.rs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::io::stdout;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result};
+use clap::{Arg, ArgMatches, Command};
+use futures::io::{self, AllowStdIo};
+
+use crate::config::Config;
+
+pub async fn main(args: &ArgMatches) -> Result<()> {
+    let config_path = args
+        .get_one::<PathBuf>("config")
+        .ok_or_else(|| anyhow!("missing config path"))?;
+    let cfg = Config::load(config_path)?;
+
+    let target = args
+        .get_one::<String>("target")
+        .ok_or_else(|| anyhow!("missing target"))?;
+    let (op, path) = cfg.parse_location(target)?;
+
+    let reader = op.reader(path).await?;
+    let mut out = AllowStdIo::new(stdout());
+    io::copy(reader, &mut out).await?;
```

Comment:
I am new to async rust, and not sure if this is the right way to copy data to stdout. If there is a better way, I am happy to update.

Reply:
We can use https://docs.rs/tokio/latest/tokio/io/fn.stdout.html#

Reply:
it seems a little tricky, if we were going to use `futures::io::copy`, the writer needs to implement `futures::AsyncWrite`, while `tokio::io::Stdout` only implments `tokio::io::AsyncWrite`. Am I missing something?

Reply:
opendal's Reader implemetns `tokio::AsyncRead` too :laughing:

Reply:
Ohh so I should use `tokio::io::copy`, right?

Reply:
> if we were going to use `futures::io::copy`

How about [`tokio::io::copy`](https://docs.rs/tokio/latest/tokio/io/fn.copy.html)?

Reply:
Thanks for your suggestion, PTAL